test/unit/execCmd.test.ts

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { EOL } from 'os';
+import { join } from 'path';
+import { expect, assert } from 'chai';
+import * as sinon from 'sinon';
+import { fs as fsCore } from '@salesforce/core';
```

Comment:
We have a chance to use a "standard" implementation like `fs-extra`, etc rather than our special one since this won't have an import cost for the executable.

Especially if we hope for this to be used beyond salesforce-owned plugins

Reply:
The core library does standardize on `fs-extra` plus some additional methods. We did this because then all of our code was pulling in core for those utilities, plus `fs-extra`, or if people didn't know to standardize on that then `fs` directly. This gave us one place to tell people to import/use.

If it is that big of a deal that we are requiring salesforce/core to do filesystem operations, then I would rather break that out into another library or remove it altogether. I don't think it is a big deal to use salesforce libraries in a non-salesforce plugin.

OR if we really want this to be used outside of salesforce plugins, then we should create an oclif lib. We couldn't do the scratch org management or salesforce project there,  but it could do the home mocking so cli executions are isolated.

```
import { cmdExec } from '@oclif/testkit`
// Obvously the names of these exports will be different, just illustrating the two libs
import { org, project } from `@salesforce/testkit`
```

Reply:
Right, I figured we'd need core for other things in the lib anyway.  The `parseJson` function on core is much nicer (and more helpful) than `JSON.parse`

Reply:
> The core library does standardize on `fs-extra` plus some additional methods. 

Core wraps graceful-fs.  fs-extra does more and works better with async.  

for example, the one from core demands a callback when I try something like 
`await fsCore.copyFile('fromHere', 'toThere);`

some data: https://nodejs.libhunt.com/compare-node-fs-extra-vs-node-graceful-fs

Reply:
Seems like the consensus is to use fsCore until proven insufficient.

Reply:
Agreed with using core until we figure it out. And I was obviously wrong about using fs-extra, which to me even illustrates more about standardization. Otherwise, every repo will use a different library for different reasons which will eventually increase binary size.